### PR TITLE
feat: add support for nested layouts

### PR DESCRIPTION
**Pull request by satazor**
BREAKING CHANGE: the API has greatly changed, please check the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,15 +68,65 @@ const About = () => (
 export default withLayout(<PrimaryLayout variant="light" />)(About);
 ```
 
-ℹ️ Layouts will receive the page to be rendered as the `children` prop.
+ℹ️ The `PrimaryLayout` component will receive the page to be rendered as the `children` prop.
+
+### Nested layouts
+
+Nested layouts are as easy as nesting them in the `withLayout`. Let's say that you have two account pages, `pages/account/profile.js` and `pages/account/settings`, and you want them to be wrapped by an `AccountLayout`. You would define the pages like so:
```
Comment (zebateira) [inline README.md]:
```suggestion
Nested layouts are as easy as nesting them in the `withLayout`. Let's say that you have two account pages, `pages/account/profile.js` and `pages/account/settings.js`, and you want them to be wrapped by an `AccountLayout`. You would define the pages like so:
```

File: README.md [hunk 2]
```diff
@@ -66,15 +68,65 @@ const About = () => (
...
+
+```js
+// pages/account/profile
+import React from 'react';
+import { withLayout } from '@moxy/next-layout';
+import { PrimaryLayout, AccountLayout } from '../components';
+import styles from './.module.css';
```
Comment (zebateira) [inline README.md]: `./profile.module.css` or `./styles.module.css`?

File: README.md [hunk 3]
```diff
@@ -66,15 +68,65 @@ const About = () => (
...
+    <PrimaryLayout />
+)(AccountProfile);
+```
+
+```js
+// pages/account/settings
+import React from 'react';
+import { withLayout } from '@moxy/next-layout';
+import { PrimaryLayout, AccountLayout } from '../components';
+import styles from './.module.css';
```
Comment (zebateira) [inline README.md]: same here

**Review by zebateira — CHANGES_REQUESTED** [review]
LGTM, just some minor docs tweaks

File: README.md (hunk 2 above)
Comment (satazor) [reply README.md]: Fixed!

File: README.md (hunk 3 above)
Comment (satazor) [reply README.md]: Fixed!

File: README.md (hunk 1 above)
Comment (satazor) [reply README.md]: Nice catch

**Comment by satazor** [comment]
@zebateira thanks for the review. I'm not sure if I should rename `withLayout` to `withLayoutTree` and `defaultLayout` to `defaultLayoutTree`. What do you think?

**Comment by zebateira** [comment]
> I'm not sure if I should rename withLayout to withLayoutTree and defaultLayout to defaultLayoutTree. What do you think?

Hm, I don't have a strong opinion on it, but I think that leaving it as is `withLayout` is ok since you can have a tree of layouts _or not_.
Plus, if you then add more nested layouts, it's something that is seamless (simply nest them), so no need to make it so explicit.

File: README.md [hunk 4]
```diff
@@ -116,73 +168,55 @@ export default App;
...
-                </CSSTransition>
-            </TransitionGroup>
-        ) }
-    </LayoutManager>
-);
+#### mapLayoutStateToLayoutTree
+
+Type: `ReactElement` or `function`
+
+In simple cases, you may defined a "static" layout tree, like so:
```
Comment (jgradim) [inline README.md]: you have defined => you may have defined

**Pull request merged**